### PR TITLE
Feature/jyxb 394 4 profile update api server action

### PR DESCRIPTION
**Pull request by yuseok-kim-edushare**
This pull request introduces several changes to enhance the profile modification functionality, including adding new API calls, improving error handling, and updating the user interface. The most important changes include the introduction of new API calls for nickname and UUID viewership, modifications to the profile modification process, and updates to the profile modification components to support image removal.

### API Enhancements:
* [`apps/client/src/action/profile/getProfileData.ts`](diffhunk://#diff-28804972e360c031c41b3904126cab837baf6c509d7f4d7afd899d397eb05419L17-R20): Added new API calls to fetch nickname and UUID viewership data. [[1]](diffhunk://#diff-28804972e360c031c41b3904126cab837baf6c509d7f4d7afd899d397eb05419L17-R20) [[2]](diffhunk://#diff-28804972e360c031c41b3904126cab837baf6c509d7f4d7afd899d397eb05419R42-R45)
* [`apps/client/src/action/profile/modifyProfileData.ts`](diffhunk://#diff-419db7fc9fead33f3972636978db4300a3d910b16ce1f6f419fc98ac96881be3R3-R10): Introduced `initializeHeaders` and `getAccessToken` for better header management and revalidation of the profile page after modification. [[1]](diffhunk://#diff-419db7fc9fead33f3972636978db4300a3d910b16ce1f6f419fc98ac96881be3R3-R10) [[2]](diffhunk://#diff-419db7fc9fead33f3972636978db4300a3d910b16ce1f6f419fc98ac96881be3R20-R21)

### Profile Modification Process:
* `apps/client/src/app/profile/modify/[id]/page.tsx`: Added user matching and redirection logic to ensure only the correct user can modify their profile. ([apps/client/src/app/profile/modify/[id]/page.tsxR3-R4](diffhunk://#diff-7af5aa09ff9794ae477bb5193f514432b516abc783368247b6725c101314a55bR3-R4), [apps/client/src/app/profile/modify/[id]/page.tsxR13-R17](diffhunk://#diff-7af5aa09ff9794ae477bb5193f514432b516abc783368247b6725c101314a55bR13-R17))
* [`apps/client/src/components/profile-modify/organisms/ProfileModifyInfo.tsx`](diffhunk://#diff-919f6f446987d2bd0e740d29d69687d6da43a8ac4086adb6dd290730cb2aa158R4-R8): Updated the profile modification logic to handle image uploads conditionally and redirect to the updated profile page. [[1]](diffhunk://#diff-919f6f446987d2bd0e740d29d69687d6da43a8ac4086adb6dd290730cb2aa158R4-R8) [[2]](diffhunk://#diff-919f6f446987d2bd0e740d29d69687d6da43a8ac4086adb6dd290730cb2aa158R33-R64) [[3]](diffhunk://#diff-919f6f446987d2bd0e740d29d69687d6da43a8ac4086adb6dd290730cb2aa158L68-R81) [[4]](diffhunk://#diff-919f6f446987d2bd0e740d29d69687d6da43a8ac4086adb6dd290730cb2aa158R92-R99)

### User Interface Updates:
* `apps/client/src/components/profile-modify/molecules/ProfileModifyAvatar.tsx` and `apps/client/src/components/profile-modify/molecules/ProfileModifyBanner.tsx`: Added support for image removal in profile modification components. [[1]](diffhunk://#diff-21821f928018cbc1ed7a1c12c60e359d3b0eb83458a86465631a5674cf41ad84R8-R15) [[2]](diffhunk://#diff-21821f928018cbc1ed7a1c12c60e359d3b0eb83458a86465631a5674cf41ad84L29-R32) [[3]](diffhunk://#diff-21821f928018cbc1ed7a1c12c60e359d3b0eb83458a86465631a5674cf41ad84R47-R52) [[4]](diffhunk://#diff-22e0382475fbc9b54cc3672c6d40fabcc6331206ec09c44bf4ac180becddac40R8-R15) [[5]](diffhunk://#diff-22e0382475fbc9b54cc3672c6d40fabcc6331206ec09c44bf4ac180becddac40R53-R56)
* [`apps/client/src/hooks/modify/useModify.ts`](diffhunk://#diff-91714bc242849dc7e4d8308232f75d7ce18ff2c1c60a5aff3658e4801570a118R141-R155): Introduced `handleImageRemove` method to handle image removal in the modification process. [[1]](diffhunk://#diff-91714bc242849dc7e4d8308232f75d7ce18ff2c1c60a5aff3658e4801570a118R141-R155) [[2]](diffhunk://#diff-91714bc242849dc7e4d8308232f75d7ce18ff2c1c60a5aff3658e4801570a118R166-R169)

### Session Management:
* [`apps/client/src/lib/api/sessionExtractor.ts`](diffhunk://#diff-52937c1ef32d2db339564cb424fd4dba64d901c6b991e5f8a04145d6afa6d07eR1-R130): Added functions to manage and validate server sessions, including token extraction and user matching.

These changes collectively improve the profile modification experience by ensuring data consistency, enhancing error handling, and providing a more intuitive user interface.

### CODE REVIEW

**Comment by yuseok-kim-edushare** [comment]
@copilot-pull-request-reviewer
right, good approach. but our api server will accept null
thus that's no matter, but good advice

**Comment by yuseok-kim-edushare** [comment]
남은 작업. 배경과 프로필 사진 없애는 기능 필요함

**Review by GwendolyNM — APPROVED** [review]
확인했습니다

File: apps/client/src/action/profile/getProfileData.ts [hunk 1]
```diff
@@ -14,7 +14,10 @@ export async function getProfileMemberInfo(id: string): Promise<ProfileMemberInf
 		headers,
 		next: { revalidate: 3600 },
 	})
-
+	void fetch(`${process.env.API_BASE_URL}/v1/profile/viewership/nickname/${id}`, {
```
Comment (yuseok-kim-edushare) [inline apps/client/src/action/profile/getProfileData.ts:17]: void 로 api fetching을 날리면 응답 안 기다리고 바로 다음 실행으로 넘어가서
조회수 단순 증가 로직은 이렇게 처리하였습니다

**Review by sing-u8 — APPROVED** [review]
확인완료

**Pull request merged**